Submitted By: Jim Gifford (jim at linuxfromscratch dot org)
Date: 2004-04-24
Initial Package Version: 0.4.8
Origin: BELFS Project
Upstream Status: Fixed in CVS
Description: Fixes Compile Issues with GCC 3.4
 
$LastChangedBy$
$Date$

diff -Naur ffmpeg-0.4.8.orig/libavcodec/i386/dsputil_mmx.c ffmpeg-0.4.8/libavcodec/i386/dsputil_mmx.c
--- ffmpeg-0.4.8.orig/libavcodec/i386/dsputil_mmx.c	2003-09-28 15:26:40.000000000 +0000
+++ ffmpeg-0.4.8/libavcodec/i386/dsputil_mmx.c	2004-04-24 18:15:55.222028167 +0000
@@ -25,14 +25,14 @@
 int mm_flags; /* multimedia extension flags */
 
 /* pixel operations */
-static const uint64_t mm_bone __attribute__ ((aligned(8))) = 0x0101010101010101ULL;
-static const uint64_t mm_wone __attribute__ ((aligned(8))) = 0x0001000100010001ULL;
-static const uint64_t mm_wtwo __attribute__ ((aligned(8))) = 0x0002000200020002ULL;
-
-static const uint64_t ff_pw_20 __attribute__ ((aligned(8))) = 0x0014001400140014ULL;
-static const uint64_t ff_pw_3  __attribute__ ((aligned(8))) = 0x0003000300030003ULL;
-static const uint64_t ff_pw_16 __attribute__ ((aligned(8))) = 0x0010001000100010ULL;
-static const uint64_t ff_pw_15 __attribute__ ((aligned(8))) = 0x000F000F000F000FULL;
+const uint64_t mm_bone __attribute__ ((aligned(8))) = 0x0101010101010101ULL;
+const uint64_t mm_wone __attribute__ ((aligned(8))) = 0x0001000100010001ULL;
+const uint64_t mm_wtwo __attribute__ ((aligned(8))) = 0x0002000200020002ULL;
+
+const uint64_t ff_pw_20 __attribute__ ((aligned(8))) = 0x0014001400140014ULL;
+const uint64_t ff_pw_3  __attribute__ ((aligned(8))) = 0x0003000300030003ULL;
+const uint64_t ff_pw_16 __attribute__ ((aligned(8))) = 0x0010001000100010ULL;
+const uint64_t ff_pw_15 __attribute__ ((aligned(8))) = 0x000F000F000F000FULL;
 
 #define JUMPALIGN() __asm __volatile (".balign 8"::)
 #define MOVQ_ZERO(regd)  __asm __volatile ("pxor %%" #regd ", %%" #regd ::)
diff -Naur ffmpeg-0.4.8.orig/libavcodec/i386/motion_est_mmx.c ffmpeg-0.4.8/libavcodec/i386/motion_est_mmx.c
--- ffmpeg-0.4.8.orig/libavcodec/i386/motion_est_mmx.c	2003-09-28 15:26:40.000000000 +0000
+++ ffmpeg-0.4.8/libavcodec/i386/motion_est_mmx.c	2004-04-24 18:15:55.226027527 +0000
@@ -20,13 +20,13 @@
  */
 #include "../dsputil.h"
 
-static const __attribute__ ((aligned(8))) uint64_t round_tab[3]={
+const __attribute__ ((aligned(8))) uint64_t round_tab[3]={
 0x0000000000000000,
 0x0001000100010001,
 0x0002000200020002,
 };
 
-static __attribute__ ((aligned(8), unused)) uint64_t bone= 0x0101010101010101LL;
+const __attribute__ ((aligned(8), unused)) uint64_t bone= 0x0101010101010101LL;
 
 static inline void sad8_mmx(uint8_t *blk1, uint8_t *blk2, int stride, int h)
 {
diff -Naur ffmpeg-0.4.8.orig/libavcodec/i386/simple_idct_mmx.c ffmpeg-0.4.8/libavcodec/i386/simple_idct_mmx.c
--- ffmpeg-0.4.8.orig/libavcodec/i386/simple_idct_mmx.c	2003-09-28 15:26:40.000000000 +0000
+++ ffmpeg-0.4.8/libavcodec/i386/simple_idct_mmx.c	2004-04-24 18:15:55.234026246 +0000
@@ -45,10 +45,10 @@
 #define ROW_SHIFT 11
 #define COL_SHIFT 20 // 6
 
-static const uint64_t __attribute__((aligned(8))) wm1010= 0xFFFF0000FFFF0000ULL;
-static const uint64_t __attribute__((aligned(8))) d40000= 0x0000000000040000ULL;
+const uint64_t __attribute__((aligned(8))) wm1010= 0xFFFF0000FFFF0000ULL;
+const uint64_t __attribute__((aligned(8))) d40000= 0x0000000000040000ULL;
 
-static const int16_t __attribute__((aligned(8))) coeffs[]= {
+const int16_t __attribute__((aligned(8))) coeffs[]= {
 	1<<(ROW_SHIFT-1), 0, 1<<(ROW_SHIFT-1), 0,
 //	1<<(COL_SHIFT-1), 0, 1<<(COL_SHIFT-1), 0,
 //	0, 1<<(COL_SHIFT-1-16), 0, 1<<(COL_SHIFT-1-16),
